static/src/javascripts/projects/common/modules/user-prefs.spec.js

```diff
@@ -0,0 +1,112 @@
+// @flow
+import userPrefs from 'projects/common/modules/user-prefs';
+import { local } from 'lib/storage';
+
+class LocalStorageMock {
+    store: Object;
+
+    constructor() {
+        this.store = {};
+    }
+
+    clear() {
+        this.store = {};
+    }
+
+    getItem(key) {
+        return this.store[key];
+    }
+
+    setItem(key, value) {
+        this.store[key] = value.toString();
+    }
+
+    removeItem(key) {
+        this.store[key] = null;
+    }
+}
+
+const localStorage = new LocalStorageMock();
+
+describe('userPrefs - Client-side preferences', () => {
+    beforeAll(() => {
+        local.storage = localStorage;
```

Comment:
It would be nicer to mock `local` using Jest's `mock` method, perhaps even moving the `LocalStorageMock` class into a [manual mock](http://facebook.github.io/jest/docs/manual-mocks.html). I'm aware that attempting to do this has caused problems for others. Have you tried this at all, @philwills?